www/_template/guides/plugins.md

```diff
@@ -4,75 +4,168 @@ title: Creating Your Own Plugin
+## Adding user-configurable options to your plugin
+TODO make this a real example
+In this step, you'll learn how to add user-configurable options to your plugin and to use them in your plugin code.
 
-See our official [@snowpack/plugin-webpack](https://github.com/snowpackjs/snowpack/tree/main/plugins/plugin-webpack) bundler plugin for an example of using the current interface.
+In your example Snowpack project, instead of enabling the plugin as a string containing the plugin name, use an array.  The first item is name of your plugin and the second a new object containing the plugin options.
 
-### Example: Getting Started
+```diff
+// snowpack.config.js
+{
+  "plugins": [
+-    "my-snowpack-plugin"
++    ["my-snowpack-plugin", { "optionA": "foo", "optionB": true }]
+  ]
+}
+```
 
-To create a Snowpack plugin, you can start with the following file template:
+You access these through the `pluginOptions`
 
-```js
+```diff
 // my-snowpack-plugin.js
 module.exports = function (snowpackConfig, pluginOptions) {
++ let optionA = pluginOptions.optionA
++ let optionB = pluginOptions.optionB
   return {
-    name: 'my-snowpack-plugin',
-    // ...
+    name: 'my-snowpack-plugin'
   };
 };
 ```
 
-```json
-// snowpack.config.json
-{
-  "plugins": [
-    ["./my-snowpack-plugin.js", { "optionA": "foo", "optionB": "bar" }]
-  ]
-}
+
+## Choosing the right hook from The Snowpack Plugin API
+The plugin API has four main hooks you'll use to integrate with Snowpack. Hooks are methods that Snowpack calls automatically when running. Choosing the right one depends on what part of the build pipeline you want your plugin to act on and what type of actions it performs:
+
+- Build plugins act on the raw files in your Snowpack project. They use the `resolve` object and `load()` hook. For example [@snowpack/plugin-sass](https://github.com/snowpackjs/snowpack/tree/main/plugins/plugin-sass) compiles SASS files to CSS files.
+- Transform plugins run on the HTML, JS, and CSS files Snowpack built from your raw file. These utilize the `transform()` hook. An example is [@snowpack/plugin-postcss](https://github.com/snowpackjs/snowpack/tree/main/plugins/plugin-postcss), which runs postcss to optimize CSS.
+- Bundler plugins replace the default [`snowpack build`](/concepts/build-pipeline) using the `optimize()` hook. An example is [@snowpack/plugin-webpack](https://github.com/snowpackjs/snowpack/tree/main/plugins/plugin-webpack) which bundles the final build using Webpack.
+- Dev tooling plugins run during `snowpack dev` OR `snowpack build` using the `run()` hook. For example [@snowpack/plugin-run-script](https://github.com/snowpackjs/snowpack/tree/main/plugins/plugin-run-script) runs any CLI command during `snowpack dev` or `snowpack build`.
+
+TODO: When do you choose a dev tooling vs. other types?
+
+## Build plugins: Using `resolve` and `load()` hook
```

Comment:
I have an idea: I think you were right to identify the opening of this guide (the intro and "getting started" sections) as needing work. But, once you get into this part of the guide (everything starting at "Choosing the right hook from The Snowpack Plugin API")  I think our current guide is actually pretty good.  Or at the very least, it's good enough as-is that we could provide more value focusing somewhere else given the number of TODOs still remaining.

I'd recommend focusing on everything that you have before "Choosing the right hook from The Snowpack Plugin API" and then deleting/reverting that section and all changes introduced after it. LMK what you think, happy to talk through this more in person.